components/accelerator/nvidia/nccl/component.go

```diff
@@ -24,28 +25,29 @@ func New(ctx context.Context, cfg nvidia_common.Config) (components.Component, e
 		return nil, err
 	}
 
-	cctx, ccancel := context.WithCancel(ctx)
-	w, err := newWatcher(cctx, eventsStore)
+	w, err := dmesg.NewWatcher()
 	if err != nil {
-		ccancel()
 		return nil, err
 	}
 
+	cctx, ccancel := context.WithCancel(ctx)
+	logLineProcessor := dmesg.NewLogLineProcessor(cctx, w, Match, eventsStore)
```

Comment:
Sorry for forgetting to left comments,  shall we create the NewWatcher in the LogLineProcessor if the w is nil?

Reply:
Oh yes, let's do that

Reply:
Done https://github.com/leptonai/gpud/pull/428